optuna/pruners/_percentile.py

```diff
@@ -167,7 +167,7 @@ def prune(self, study, trial):
             return False
 
         n_warmup_steps = self._n_warmup_steps
-        if step <= n_warmup_steps:
+        if step <= n_warmup_steps - 1:
```

Comment:
Could you simplify the condition as follows?
```suggestion
        if step < n_warmup_steps:
```